### PR TITLE
BUG: fix test_arrow.py tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
Sits on top of #48428

### CODE REVIEW

File: pandas/core/internals/concat.py [hunk 1]
```diff
@@ -471,7 +463,8 @@ def get_reindexed_values(self, empty_dtype: DtypeObj, upcasted_na) -> ArrayLike:
                     if len(values) and values[0] is None:
                         fill_value = None
 
-                if is_datetime64tz_dtype(empty_dtype):
```
Comment (mroeschke) [inline pandas/core/internals/concat.py:474]: Technically, these checks would be okay to keep with https://github.com/pandas-dev/pandas/pull/48258 changed?

Comment (jbrockmendel) [reply pandas/core/internals/concat.py:474]: yes

Comment (jbrockmendel) [reply pandas/core/internals/concat.py:474]: I'm fine with waiting on 48258, but also think in general `isinstance(obj.dtype, DatetimeTZDtype)` is a better pattern than `is_datetime64tz_dtype(obj.dtype)` as it is more explicit/performant.  Main downside is having two different ways of doing the same check

Comment (mroeschke) [reply pandas/core/internals/concat.py:474]: Okay that sounds like good reasoning to me to move forward with this PR first. I'm not much closer on debugging the seemingly unrelated failures with https://github.com/pandas-dev/pandas/pull/48258 so don't want to block.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**